src/main/scala/com/snowplowanalytics/snowplow/rdbloader/LoaderError.scala

```diff
@@ -34,6 +33,35 @@ object LoaderError {
     }
   }
 
+  /** Overloaded method to surface helpful error messages
+    * in case a [[RawConfig]] cannot be successfully
+    * transformed into a [[CliConfig]].
+    */
+  object ErrorSurface {
+    object ValidationStep extends Enumeration {
+      type ValidationStep = Value
+      val logkeyValidation, folderValidation, resolverValidation, targetValidation = Value
+    }
+    import ValidationStep._
+
+    def surface(error: ConfigError): ConfigError = error match {
+      case ParseError(message) =>
+        ParseError("Error trying to base64-decode the <config.yml> file passed to `--config`. Underlying error message:\n" + message)
```

Comment:
Also, I'd recommend to not use `+` for string concatenation as it always falls back to Java `toString` and if one of your arguments is not actually a string, you can get unexpected results:

```scala
@ "foo" + List(1,2,3)
res1: String = "fooList(1, 2, 3)"
```

`++` will give something even less expected:

```scala
@ "foo" ++ List(1,2,3)
res2: collection.immutable.IndexedSeq[AnyVal] = Vector('f', 'o', 'o', 1, 2, 3)
```

But at least its return type will warn you about it.

Reply:
Ah, I'm actually not a big fan of it myself, but that is how it's done elsewhere in the code and I wanted to keep it consistent.